### ISSUE TITLE
Application resource plans hitting RequiresReplace can fail with "application already exists"

### ISSUE THREAD

**Issue by hmlanigan** [juju/terraform-provider-juju#272]
### Description

Depending on speed of hardware, plan changes requiring a destroy of an application and new creation, such as changing constraints will fail the first time as the application is not removed from juju fast enough.

However waiting a little bit after the failure, for the application to be removed, and running the plan a 2nd time will succeed in creating the new application.

### Urgency

Annoying bug in our test suite

### Terraform Juju Provider version

0.8.0

### Terraform version

1.5.5

### Terraform Configuration(s)

```terraform
terraform {
  required_providers {
    juju = {
      version = ">= 0.7.0"
      source  = "juju/juju"
    }
  }
}

provider "juju" {
}

resource "juju_model" "this" {
  name = "unit-test"
}

resource "juju_application" "this" {
  model = juju_model.this.name
  name = "ubuntu-lite"
  charm {
    name = "jameinel-ubuntu-lite"
  }
  trust = true
  expose{}
}
```

### Reproduce / Test

```shell
With the above plan, run:
`terraform init && terraform plan && terraform apply`

Once deploy is finished, update the plan to include in the application resource:
`constraints = "arch=amd64 cores=1 mem=4096M"`

Now run `terraform plan && terraform apply`, the plan should be:
`Plan: 1 to add, 0 to change, 1 to destroy.`

Type `yes` to run the plan. Depending on the speed of the hardware used, this should trigger:

juju_application.this: Destroying... [id=unit-test:ubuntu-lite]
juju_application.this: Destruction complete after 0s
juju_application.this: Creating...
╷
│ Error: Client Error
│
│   with juju_application.this,
│   on heather.tf line 17, in resource "juju_application" "this":
│   17: resource "juju_application" "this" {
│
│ Unable to create application, got error: cannot add application "ubuntu-lite": application already exists
```
```


### Debug/Panic Output

_No response_

### Notes & References

_No response_

**Comment by hmlanigan** [comment]
There may still be scenarios where this is a problem. However the majority of cases should be fixed. Open a new issue if this occurs again.
